public/apps/configuration/top-nav-menu.tsx

```diff
@@ -59,6 +60,8 @@ export const SecurityPluginTopNavMenu = React.memo(
               : undefined,
           onSelectedDataSources: wrapSetDataSourceWithUpdateUrl,
           fullWidth: true,
+          dataSourceFilter: (ds) =>
+            ds.attributes.auth.credentials?.service !== SigV4ServiceName.OpenSearchServerless,
```

Comment:
Can we add a test to verify this behavior ?

Reply:
I saw you comment above. We are relying on external tests. Instead is there a way to add a mock and test it in our tests?

Reply:
IMHO it is crossing the testing boundary. The same way we don't test that OUI tables filter or sort properly we should expect that this component provided by core does the same. An additional layer of complexity here is that we would need to assume the network call made by the component, and if that changes in the future (among other implementation changes), it would break the test. WDYT?

Reply:
If there is at-least 1 change in core testing this behavior then I'm okay with merging this change without tests